pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -264,9 +264,23 @@ func (s *Service) getOrCreateAPILoadBalancer(openStackCluster *infrav1.OpenStack
 		return nil, fmt.Errorf("network is not yet available in OpenStackCluster.Status")
 	}
 
-	// Create the VIP on the first cluster subnet
-	subnetID := openStackCluster.Status.Network.Subnets[0].ID
-	s.scope.Logger().Info("Creating load balancer in subnet", "subnetID", subnetID, "name", loadBalancerName)
+	var vipNetworkID, vipSubnetID string
+	if openStackCluster.Status.APIServerLoadBalancer != nil {
+		if openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork != nil {
+			if openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.ID != "" {
+				vipNetworkID = openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.ID
+			}
+			if len(openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets) > 0 {
+				vipSubnetID = openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets[0].ID
```

Comment:
Looks like we assume there can only be one subnet. Let's make sure to document that on the CRD, that only first element is taken into account. Because I assume we don't support dual stack LBs yet in any case, right?

Reply:
well the additionalVips are still in beta in gophercloud and only part of  `v2.0.0-beta.1` (or `v2.0.0-beta.2`), I suppose we don't want to upgrade to that, right?

Reply:
Yes, I'd rather not upgrade or add support for additionalVIPs here, but I'm really glad the API design assumes it will happen eventually.